ice/cli/websocket/client.zep

```diff
@@ -0,0 +1,210 @@
+
+namespace Ice\Cli\Websocket;
+
+use Ice\Exception;
+
+/**
+ * Websocket client.
+ *
+ * @package     Ice/Cli
+ * @category    Component
+ * @author      Ice Team
+ * @copyright   (c) 2014-2018 Ice Team
+ * @license     http://iceframework.org/license
+ */
+class Client extends Websocket
+{
+
+    protected socket = null;
+    protected message = null;
+    protected tick = null;
+
+    /**
+     * Connect to server.
+     *
+     * @param string address Address to bind to, defaults to `ws://127.0.0.1:8080`
+     * @param array headers Optional array of headers to pass when connecting
+     * @return self
+     */
+    public function connect(string address = "ws://127.0.0.1:8080", var headers = []) -> <self>
+    {
+        var addr, key, name, value, res, data, matches;
+
+        let addr = parse_url(address);
+
+        if addr === false || !isset addr["host"] || !isset addr["port"] {
+            throw new Exception("Invalid address");
+        }
+
+        let this->socket = fsockopen(
+            (isset addr["scheme"] && in_array(addr["scheme"], ["ssl", "tls", "wss"]) ? "tls://" : "") . addr["host"],
+            addr["port"]
+        );
+
+        if this->socket === false {
+            throw new Exception("Could not connect");
+        }
+
+        let key = this->generateKey(),
+            headers = array_merge(
+                this->normalizeHeaders([
+                    "Host": addr["host"] . ":" . addr["port"],
+                    "Connection": "Upgrade",
+                    "Upgrade": "websocket",
+                    "Sec-Websocket-Key": key,
+                    "Sec-Websocket-Version": "13"
+                ]),
+                this->normalizeHeaders(headers)
+            );
+
+        let key = headers["Sec-Websocket-Key"];
+
+        for name, value in headers {
+            let headers[name] = name . ": " . value;
+        }
+
+        if isset addr["path"] && strlen(addr["path"]) {
```

Comment:
why not `!empty addr["path"]`

Reply:
I just noticed `empty` doesn't work like in PHP https://github.com/phalcon/zephir/issues/917.